Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -312,11 +316,118 @@ public async Task Submit(SubmitSubmissionServiceModel model)
             await this.submissionsData.Add(newSubmission);
             await this.submissionsData.SaveChanges();
 
-            // newSubmission.Problem = problem;
-            // newSubmission.SubmissionType = await this.submissionTypesBusinessService.GetById(newSubmission.SubmissionTypeId.Value);
+            newSubmission.Problem = problem;
+
+            var response = this.submissionsDistributorCommunicationService.AddSubmissionForProcessing(newSubmission).Result;
+        }
+
+        public void ProcessExecutionResult(SubmissionExecutionResult submissionExecutionResult)
+        {
+            var submission = this.submissionsData
+                .GetByIdQuery(submissionExecutionResult.SubmissionId)
+                .Include(s => s.Problem.Tests)
+                .FirstOrDefault();
+
+            if (submission == null)
+            {
+                throw new BusinessServiceException(
+                    $"Submission with Id: \"{submissionExecutionResult.SubmissionId}\" not found.");
+            }
+
+            var exception = submissionExecutionResult.Exception;
+            var executionResult = submissionExecutionResult.ExecutionResult;
+
+            submission.ProcessingComment = null;
+            this.testRunsDataService.DeleteBySubmission(submission.Id);
+
+            if (executionResult != null)
+            {
+                this.ProcessTestsExecutionResult(submission, executionResult);
+            }
+            else if (exception != null)
+            {
+                submission.ProcessingComment = exception.Message + Environment.NewLine + exception.StackTrace;
+            }
+            else
+            {
+                submission.ProcessingComment = "Invalid execution result received. Please contact an administrator.";
+            }
+        }
```

Comment:
This can be improved to decrease the nesting, if the order is not strict.

```
if(exception != null) {
  ...
  return;
}

if (executionResult == nul) {
  ...
  return;
}

this.ProcessTestsExecutionResult(...);

```